server/base/src/main/java/org/apache/accumulo/server/conf/store/impl/ZooPropStore.java

```diff
@@ -172,8 +172,8 @@ public void create(PropStoreKey<?> propStoreKey, Map<String,String> props) {
    */
   @Override
   public @NonNull VersionedProperties get(final PropStoreKey<?> propStoreKey) {
-    checkZkConnection(); // if ZK not connected, block, do not just return a cached value.
     propStoreWatcher.registerListener(propStoreKey, this);
+    checkZkConnection(); // if ZK not connected, block, do not just return a cached value.
```

Comment:
This seems to be a good change.

Reply:
I needed to move this line down for my test to work because I was closing the zookeeper server connection in `checkZKConnection` and I needed the watcher to notice the connection closed event. I'm not sure if this is helpful, hurtful, or benign in a non-test scenario.